src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Tables/Resource.sql

```diff
@@ -5,16 +5,15 @@
     Version                     int                     NOT NULL,
     IsHistory                   bit                     NOT NULL,
     ResourceSurrogateId         bigint                  NOT NULL,
-    CONSTRAINT UQ_Resource_ResourceSurrogateId UNIQUE (ResourceSurrogateId)
-    ON [Primary],
-    CONSTRAINT PKC_Resource PRIMARY KEY CLUSTERED (ResourceTypeId, ResourceSurrogateId)
-    WITH (DATA_COMPRESSION = PAGE)
-    ON PartitionScheme_ResourceTypeId(ResourceTypeId),
     IsDeleted                   bit                     NOT NULL,
     RequestMethod               varchar(10)             NULL,
     RawResource                 varbinary(max)          NOT NULL,
     IsRawResourceMetaSet        bit                     NOT NULL DEFAULT 0,
-    SearchParamHash             varchar(64)             NULL
+    SearchParamHash             varchar(64)             NULL,
+
+    CONSTRAINT PKC_Resource PRIMARY KEY CLUSTERED (ResourceTypeId, ResourceSurrogateId)
```

Comment:
>ResourceTypeId, ResourceSurrogateId) [](http://example.com/codeflow?start=51&length=36)

type, id
1, 1
2,1
2,1

first index prevents record 2 and 3 to be inserted.
second index only record 3

they not completely interchangeable...

Reply:
Unique constraint(UQ_Resource_ResourceSurrogateId) also creates index on resourceSurrogateId internally by SqlServer and  since we already have unique index on ResourceSurrogateId(UQIX_Resource_ResourceSurrogateId ), we don't need unique constraint anymore. Does that makes sense?